athena/graph_builder/contact_graph_builder.py

```diff
@@ -28,55 +23,108 @@ def dilation(args) -> list:
     return [(obj, cell, {EDGE_WEIGHT: 1}) for cell in cells]
 
 
-# %%
 class ContactGraphBuilder(BaseGraphBuilder):
-    '''Contact-Graph class.
+    """Contact-Graph class.
 
     Build contact graph based on pixel expansion of cell masks.
-    '''
+    """
 
     def __init__(self, config: dict):
         """Base-Graph Builder constructor
 
         Args:
             config: Dictionary containing a dict called `builder_params` that provides function call arguments to the build_topology function
+
+        Default config:
+            config = {'builder_params':
+                    {'dilation_kernel': 'disk',
+                     'radius': 4,
+                     'include_self':False},
+                'concept_params':
+                    {'filter_col':None,
+                    'labels':None},
+                'coordinate_keys': ['x', 'y'],
+                'mask_key': 'cellmasks'}
         """
+        self.builder_type = "contact"
         super().__init__(config)
 
-    def _build_topology(self, topo_data: dict, **kwargs) -> None:
+    def __call__(self, so, spl):
         """Build topology using pixel expansion of segmentation masks provided by topo_data['mask']. Masks that overlap after expansion are connected in the graph.
 
         Args:
-            topo_data: dict providing the segmentation mask under key 'mask'
+            so: spatial omic object
+            spl: sting identifying the sample in the spatial omics object
 
         Returns:
+            Graph and key to graph in the spatial omics object
 
         """
 
-        # type hints
-        self.graph: nx.Graph
-
-        params = self.config['builder_params']
+        # Unpack parameters for building
+        if self.config["build_concept_graph"]:
+            filter_col = self.config["concept_params"]["filter_col"]
+            labels = self.config["concept_params"]["labels"]
```

Comment:
So the concept graph builder relies on a column 'labels' in obs? Maybe raise an error if this is not present

Reply:
Rather `config["concept_params"]["filter_col"]` is a field in the config that specifies the column in `so.obs[spl]` that one wishes to use in the concept graph building process. Namely, the labels present in that column will serve as criteria for either including or excluding cells from the graph. The filed `config["concept_params"]["lables"]` specifies which labels/values in column `config["concept_params"]["filter_col"]` should be INCLUDED in the graph. For example if `filter_col` is `"cell _type"` then `lables` is `["tumor", "immune"]`. 

There are already checks that raise an exception if the column or the labels specified are not in `so.obs[spl]`. Link to [function](https://github.com/Sann5/ATHENA/blob/4ed25d69de3a48af5934de99fbce25d4b883b3e9/athena/graph_builder/base_graph_builder.py#L48) in `BaseGraphBuilder`.